### PR TITLE
Riskmgmtv1

### LINKED ISSUE #17: Initial Version of Risk Mgmt module
Possibility of jumping orders; There is no enforcement of manager to take best offered order

- Initial solution:
Assert that: 
`order.owner === address(liquidityProvider)`
for takeOrder

### PR DESCRIPTION
**Pull request by retotrinkler**
Closes #17

### CODE REVIEW

File: contracts/Core.sol [hunk 1]
```diff
@@ -48,27 +51,22 @@ contract Core is DBC, Owned, Shares, SafeMath, CoreProtocol {
...
-    uint public sharePrice;
-    uint public unclaimedFees; // Combined total of Management and Performance fees earned by portfolio manager
     // Fields that are only changed in constructor
     address referenceAsset;
     address melonAsset;
     // Fields that can be changed by functions
     Modules module;
     Properties properties;
-    Calculations calculated;
+    Calculations public atLastPayout;
```
Comment (travs) [inline contracts/Core.sol]: Good variable name, more clear to me now :+1:

File: contracts/Core.sol [hunk 2]
```diff
@@ -259,18 +269,17 @@ contract Core is DBC, Owned, Shares, SafeMath, CoreProtocol {
 
     /// Pre: Universe has been defined
     /// Post: Whether buying and selling of tokens are allowed at given exchange
-    function isWithinKnownUniverse(address onExchange, address sell_which_token, address buy_which_token)
+    function requireIsWithinKnownUniverse(address onExchange, address sell_which_token, address buy_which_token)
```
Comment (travs) [inline contracts/Core.sol]: Not a big deal since it works as is, but should this be a modifier I wonder?

File: test/makeOrder.js [hunk 3]
```diff
@@ -161,11 +161,11 @@ contract('Net Asset Value', (accounts) => {
       });
     });
 
-    it('Create an initial order', (done) => {
+    it.skip('Create an initial order', (done) => {
       const wantedShares = new BigNumber(2e+17);
       const wantedValue = new BigNumber(2e+17);
       const expectedValue = new BigNumber(2e+17);
-
+
```
Comment (travs) [inline test/makeOrder.js]: oops

File: contracts/Core.sol (hunk 1 above)
Comment (retotrinkler) [inline contracts/Core.sol]: :)

File: contracts/Core.sol (hunk 2 above)
Comment (retotrinkler) [inline contracts/Core.sol]: ideally yes! however there is [this one time](https://github.com/melonproject/protocol/blob/c1c3a8d670fbcf2402528eb90e96c9d8e44ac0c2/contracts/Core.sol#L244-L249) where it needs to get (call) the parameteres for it first

File: test/makeOrder.js (hunk 3 above)
Comment (retotrinkler) [inline test/makeOrder.js]: ah np at all - guess it happened automatically w my tool

**Pull request merged**